### PR TITLE
fix: Update file api

### PR DESCRIPTION
**Pull request by kevjumba**
**What this PR does / why we need it**:
Change file options param file_url to uri. Deprecates file_url immediately because file_url is not user facing and is only used in FileSource.
DOES NOT update the API for java(i.e java still uses fileurl).

**Which issue(s) this PR fixes**:

Fixes #

### CODE REVIEW

File: protos/feast/core/DataSource.proto [hunk 1]
```diff
@@ -89,10 +89,18 @@ message DataSource {
     // s3://path/to/file for AWS S3 storage
     // gs://path/to/file for GCP GCS storage
     // file:///path/to/file for local storage
+    // Deprecated in Feast 0.23.0
     string file_url  = 2;
```
Comment (achals) [inline protos/feast/core/DataSource.proto:92]: You should change this in place; serialized protobufs only use the index number not the field name.

**Review by achals — CHANGES_REQUESTED** [review]

File: protos/feast/core/DataSource.proto (hunk 1 above)
Comment (kevjumba) [reply protos/feast/core/DataSource.proto:92]: Ah I didn't know that but that makes a lot of sense! Thanks for the suggestion that makes the update so much easier.

File: sdk/python/feast/infra/offline_stores/file_source.py [hunk 2]
```diff
@@ -180,19 +179,20 @@ class FileOptions:
     def __init__(
         self,
         file_format: Optional[FileFormat],
-        file_url: Optional[str],
         s3_endpoint_override: Optional[str],
+        uri: Optional[str],
```
Comment (achals) [inline sdk/python/feast/infra/offline_stores/file_source.py:183]: Wanted to confirm that this is not user facing?

**Review by achals — COMMENTED** [review]
just a question, but looks good otherwise.

File: sdk/python/feast/infra/offline_stores/file_source.py (hunk 2 above)
Comment (kevjumba) [reply sdk/python/feast/infra/offline_stores/file_source.py:183]: Yup isn't called anymore but maybe @adchia can also confirm?

File: sdk/python/feast/infra/offline_stores/file_source.py [hunk 3]
```diff
@@ -182,17 +183,31 @@ def __init__(
...
+                (
+                    "The option to pass a file_url parameter to FileOptions is being deprecated. "
+                    "Please pass the file url to the uri parameter instead. The parameter will be deprecated in Feast 0.23"
+                ),
+                DeprecationWarning,
+            )
         self._file_url = file_url
+        self._uri = uri
+        if not self._uri:
+            self._uri = self._file_url
```
Comment (achals) [inline sdk/python/feast/infra/offline_stores/file_source.py:210]: you can do `self._uri = uri or file_url` and probably not store `self._file_url` at all

Comment (kevjumba) [reply sdk/python/feast/infra/offline_stores/file_source.py:210]: Great catch! thanks!

**Review by achals — APPROVED** [review]
/lgtm

**Pull request merged**